src/ray/object_manager/object_manager.cc

```diff
@@ -344,6 +349,15 @@ void ObjectManager::Push(const ObjectID &object_id, const NodeID &node_id) {
         nodes.emplace(node_id, std::move(timer));
       }
     }
+    // If the object was spilled on this node, we should try pulling it.
+    // In this case, the object will be pulled from the external storage in a local node.
+    if (is_object_spilled_locally_(object_id)) {
```

Comment:
This is essential because HandlePull assumes that the object already exists in the local node. In object spilling case, it is not the case, so we should have another pull here.